db/migrations/20220510112000_add_application_constraint.go

```diff
@@ -0,0 +1,39 @@
+package migrations
+
+import (
+	logging "github.com/RedHatInsights/sources-api-go/logger"
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+func AddApplicationConstraint() *gormigrate.Migration {
+	type Application struct {
+		gorm.Model
+		SourceID          int64 `gorm:"uniqueIndex:index_applications_on_source_id_tenant_id_app_type_id"`
+		ApplicationTypeID int64 `gorm:"uniqueIndex:index_applications_on_source_id_tenant_id_app_type_id"`
+		TenantID          int64 `gorm:"uniqueIndex:index_applications_on_source_id_tenant_id_app_type_id"`
```

Comment:
I would say to stick to how [Postgres names its indexes](https://stackoverflow.com/a/4108266). So that would make the index name `applications_app_type_id_source_id_tenant_id_idx`.

Reply:
well I followed what we already have in db ... e.g. these are our existing indexes names in applications table 
![image](https://user-images.githubusercontent.com/89980168/168595238-75e16454-77a5-43a5-b0eb-3988397a5eba.png)
 
My first option was `"idx_xxxxx"` because that is how it is used in [documentation](https://gorm.io/docs/indexes.html), but then I decided to let it in same way how we have other existing indexes

Reply:
I see, but there's a couple of things:

* We still have that naming because the PR which transforms all those names is still pending to be merged: https://github.com/RedHatInsights/sources-api-go/pull/175
* I chose Postgres naming because that's standard to any user that uses Postgres, instead of framework specific like it was with Active Record, or like it is with Gorm.

Anyway we can merge as it is, and I'll update my PR afterwards.

Reply:
I see 😁  thanks for the reminder, i completely forgot we have these pending PRs ... I will set same naming here 👍🏻